NAS driver crashed when provisioning a block volume

I tried to create a block volume using the NAS provisioner and the driver crashed. It should not crash, it must return an error that block volumes are not supported.

```
time="2021-08-10T00:48:31+08:00" level=info msg="CreateVolume: Starting NFS CreateVolume, nas-955a70f0-1200-4c0c-bff4-a04a6ec2931d, name:\"nas-955a70f0-1200-4c0c-bff4-a04a6ec2931d\" capacity_range:<required_bytes:1073741824 > volume_capabilities:<block:<> access_mode:<mode:SINGLE_NODE_WRITER > > parameters:<key:\"archiveOnDelete\" value:\"false\" > parameters:<key:\"server\" value:\"112b624b359-ndi10.eu-central-1.nas.aliyuncs.com\" > parameters:<key:\"volumeAs\" value:\"subpath\" > "
panic: interface conversion: csi.isVolumeCapability_AccessType is *csi.VolumeCapability_Block, not *csi.VolumeCapability_Mount
goroutine 95 [running]:
github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/nas.(*controllerServer).CreateVolume(0xc0006c2840, 0x234e9a0, 0xc00090bd10, 0xc00031e7e0, 0xc0006c2840, 0x1, 0xc00075d2a0)
        /home/regressionTest/go/src/github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/nas/controllerserver.go:185 +0x6479
github.com/container-storage-interface/spec/lib/go/csi._Controller_CreateVolume_Handler.func1(0x234e9a0, 0xc00090bd10, 0x1f22ae0, 0xc00031e7e0, 0x18, 0xc00075d280, 0x7f7fa0e4d3a8, 0x30)
        /home/regressionTest/go/pkg/mod/github.com/container-storage-interface/spec@v1.1.0/lib/go/csi/csi.pb.go:4947 +0x89
github.com/kubernetes-csi/drivers/pkg/csi-common.logGRPC(0x234e9a0, 0xc00090bd10, 0x1f22ae0, 0xc00031e7e0, 0xc00075d1e0, 0xc00075d280, 0xc0001b6b78, 0x508cc6, 0x1eb95a0, 0xc00090bd10)
        /home/regressionTest/go/pkg/mod/github.com/kubernetes-csi/drivers@v1.0.2/pkg/csi-common/utils.go:99 +0x15d
github.com/container-storage-interface/spec/lib/go/csi._Controller_CreateVolume_Handler(0x1efb880, 0xc0006c2840, 0x234e9a0, 0xc00090bd10, 0xc0008ff740, 0x218d300, 0x234e9a0, 0xc00090bd10, 0xc0003da210, 0xa4)
        /home/regressionTest/go/pkg/mod/github.com/container-storage-interface/spec@v1.1.0/lib/go/csi/csi.pb.go:4949 +0x150
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00012d800, 0x236c7c0, 0xc00090ec00, 0xc0000c0e00, 0xc00068ec60, 0x3546800, 0x0, 0x0, 0x0)
        /home/regressionTest/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1024 +0x522
google.golang.org/grpc.(*Server).handleStream(0xc00012d800, 0x236c7c0, 0xc00090ec00, 0xc0000c0e00, 0x0)
        /home/regressionTest/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1313 +0xd34
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00066c1c0, 0xc00012d800, 0x236c7c0, 0xc00090ec00, 0xc0000c0e00)
        /home/regressionTest/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:722 +0xa5
created by google.golang.org/grpc.(*Server).serveStreams.func1
```

StorageClass:
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
provisioner: nasplugin.csi.alibabacloud.com
metadata:
  name: alicloud-nas-subpath
mountOptions:
- nolock,tcp,noresvport
- vers=3
parameters:
  volumeAs: subpath
  server: "*censored*"
  archiveOnDelete: "false"
```

PVC:
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
    annotations:
      volume.beta.kubernetes.io/storage-provisioner: nasplugin.csi.alibabacloud.com
    name: nasplugin.csi.alibabacloud.comhbzz5
spec:
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 1Gi
    storageClassName: alicloud-nas-subpath
    volumeMode: Block
```

--- a/build/build-all.sh
+++ b/build/build-all.sh
@@ -15,6 +15,7 @@ cp build/oss/csiplugin-connector.go build/ack/csiplugin-connector.go
 cp build/oss/csiplugin-connector.service build/ack/csiplugin-connector.service
 cp build/oss/ossfs_1.80.6_centos7.0_x86_64.rpm build/ack/ossfs_1.80.6_centos7.0_x86_64.rpm
 cp build/oss/nsenter build/ack/nsenter
+cp build/oss/jindofs-fuse build/ack/jindofs-fuse
 
 export GOARCH="amd64"
 export GOOS="linux"
@@ -32,7 +33,7 @@ CGO_ENABLED=0 go build csiplugin-connector.go
 if [ "$1" == "" ]; then
   mv ${GOPATH}/src/github.com/$REPO_NAME/alibaba-cloud-csi-driver/plugin.csi.alibabacloud.com ./
   docker build -t=registry.cn-hangzhou.aliyuncs.com/acs/csi-plugin:$VERSION-$GIT_HASH ./
-  rm -rf csiplugin-connector.go csiplugin-connector.service csiplugin-connector plugin.csi.alibabacloud.com ossfs_1.80.6_centos7.0_x86_64.rpm nsenter
+  rm -rf csiplugin-connector.go csiplugin-connector.service csiplugin-connector plugin.csi.alibabacloud.com ossfs_1.80.6_centos7.0_x86_64.rpm nsenter jindofs-fuse
   docker push registry.cn-hangzhou.aliyuncs.com/acs/csi-plugin:$VERSION-$GIT_HASH
 fi
 

--- a/pkg/cpfs/controllerserver.go
+++ b/pkg/cpfs/controllerserver.go
@@ -91,7 +91,11 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	pvName := req.Name
 	cpfsOptions := []string{}
 	for _, volCap := range req.VolumeCapabilities {
-		volCapMount := ((*volCap).AccessType).(*csi.VolumeCapability_Mount)
+		var volCapMount *csi.VolumeCapability_Mount
+		volCapMount, ok := ((*volCap).AccessType).(*csi.VolumeCapability_Mount)
+		if !ok {
+			return nil, fmt.Errorf("CreateVolume: Input cpfs type error: volume: %v", req)
+		}
 		for _, mountFlag := range volCapMount.Mount.MountFlags {
 			cpfsOptions = append(cpfsOptions, mountFlag)
 		}

--- a/pkg/nas/controllerserver.go
+++ b/pkg/nas/controllerserver.go
@@ -184,7 +184,10 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	pvName := req.Name
 	nfsOptions := []string{}
 	for _, volCap := range req.VolumeCapabilities {
-		volCapMount := ((*volCap).AccessType).(*csi.VolumeCapability_Mount)
+		volCapMount, ok := ((*volCap).AccessType).(*csi.VolumeCapability_Mount)
+		if !ok {
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid accessType of create volumes: %v", volCap)
+		}
 		for _, mountFlag := range volCapMount.Mount.MountFlags {
 			nfsOptions = append(nfsOptions, mountFlag)
 		}

